controls/V-41670.rb

```diff
@@ -1,46 +1,96 @@
+  might contain communication ports, protocols, services, trust relationships,
+  user names, etc.
+
+      The web server must protect the log data from unauthorized read, write,
+  copy, etc. This can be done by the web server if the web server is also doing
+  the logging function. The web server may also use an external log system. In
+  either case, the logs must be protected from access by non-privileged users.
+  "
+  desc  "check", "
+  Review the NGINX web server documentation and deployed configuration settings to
+  determine if the web server logging features protect log information from
+  unauthorized access.
+
+  Check for the following: 
+      # grep for 'access_log' and 'error_log' directives in the nginx.conf 
+      and any separated include configuration file.
+
+  Execute the following commands:
+      # ls -alH <nginx log directory>
+      # ls -alH <path to access_log>/access.log
+      # ls -alH <path to error_log>/error.log
+  
+  Note the owner and group permissions on these files. Only system administrators 
+  and service accounts running the server should have permissions to the directory and files.
+     - The SA or service account should own the directory and files
+     - Permissions on the directory should be 750 or more restrictive
+     - Permissions on these files should be 660 or more restrictive
+  
+  If any users other than those authorized have read access to the log files, this 
+  is a finding.
   "
+  desc  "fix", "To protect the integrity of the data that is being captured in the 
+  log files, ensure that only the members of the Auditors group, Administrators, 
+  and the user assigned to run the web server software is granted permissions to 
+  read the log files."
+
   impact 0.5
+  tag "severity": "medium"
   tag "gtitle": "SRG-APP-000118-WSR-000068"
   tag "gid": "V-41670"
   tag "rid": "SV-54247r3_rule"
   tag "stig_id": "SRG-APP-000118-WSR-000068"
   tag "fix_id": "F-47129r2_fix"
   tag "cci": ["CCI-000162"]
   tag "nist": ["AU-9", "Rev_4"]
-  tag "false_negatives": nil
-  tag "false_positives": nil
-  tag "documentable": false
-  tag "mitigations": nil
-  tag "severity_override_guidance": false
-  tag "potential_impacts": nil
-  tag "third_party_tools": nil
-  tag "mitigation_controls": nil
-  tag "responsibility": nil
-  tag "ia_controls": nil
-  tag "check": "Review the web server documentation and deployed configuration
-settings to determine if the web server logging features protect log
-information from unauthorized access.
-
-Review file system settings to verify the log files have secure file
-permissions.
-
-If the web server log files are not protected from unauthorized access, this is
-a finding."
-  tag "fix": "Configure the web server log files so unauthorized access of log
-information is not possible."
+
+  authorized_sa_user_list = input('sys_admin').clone << input('nginx_owner')
+  authorized_sa_group_list = input('sys_admin_group').clone << input('nginx_group')
+
```

Comment:
call inputs directly

Reply:
nevermind you are but creating a local instance variable with a combined data set - this is fine